source/fundamentals/builders/aggregates.txt

```diff
@@ -11,3 +11,506 @@ Aggregates Builders
+   :start-after: // begin basic lookup
+   :end-before: // end basic lookup
+   :language: java
+   :dedent:
+
+Full Join and Uncorrelated SubQueries
+++++++++++++++++++++++++++++++++++++++
+
+The following example creates a pipeline stage that joins two collections, ``orders``
+and ``warehouses``, by the item and whether the available quantity is enough
+to fulfill the ordered quantity:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+   :start-after: // begin advanced lookup
+   :end-before: // end advanced lookup
+   :language: java
+   :dedent:
+
+Group
+-----
+
+Use the ``group()`` method to create a :manual:`$group </reference/operator/aggregation/group/>`
+pipeline stage to group documents by a specified expression and output a document
+for each distinct grouping.
+
+.. tip::
+
+   The driver includes the :java-core-api:`Accumulators <com/mongodb/client/model/Accumulators.html>`
+   class with static factory methods for each of the supported accumulator.
+
+The following example creates a pipeline stage that groups documents by the value
+of the ``customerId`` field. Each group accumulates the sum and average
+of the values of the ``quantity`` field into the ``totalQuantity`` and
+``averageQuantity`` fields.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+   :start-after: // begin group
+   :end-before: // end group
+   :language: java
+   :dedent:
+
+Unwind
+------
+
+Use the ``unwind()`` method to create an :manual:`$unwind </reference/operator/aggregation/unwind/>`
+pipeline stage to deconstruct an array field from input documents, creating
+an output document for each array element.
+
+The following example creates a document for each element in the ``sizes`` array:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+   :start-after: // begin unwindStage
+   :end-before: // end unwindStage
+   :language: java
+   :dedent:
+
+To preserve documents that have missing or ``null``
+values for the array field, or where array is empty:
+
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+   :start-after: // begin unwindPreserve
+   :end-before: // end unwindPreserve
+   :language: java
+   :dedent:
+
+To include the array index, in this example in a field called ``"position"``:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+   :start-after: // begin unwindIndex
+   :end-before: // end unwindIndex
+   :language: java
+   :dedent:
+
+Out
+---
+
+Use the ``out()`` method to create an :manual:`$out </reference/operator/aggregation/out/>`
+pipeline stage that writes all documents to the specified collection in
+the same database.
+
+.. tip::
+
+   The ``$out`` stage must be the last stage in any aggregation pipeline.
+
+The following example writes the pipeline to the ``authors`` collection:
```

Comment:
It doesn't really write the pipeline.  It writes the results of executing the pipeline.  Consider re-wording.

Reply:
Sorry, I forgot to include "results of the". Thanks for the catch.

Reply:
FYI, the requested change hasn't been made yet.

Reply:
Thanks, I've updated it!